### ISSUE TITLE
.md files are bugged, look terrible

### ISSUE THREAD

**Issue by vibonacci** [dracula/notepad-plus-plus#24]
![image](https://user-images.githubusercontent.com/9674553/62996763-1c2a8100-be66-11e9-9dfe-9639a026fafc.png)

As you can see in the above example, .md files look very strange. When you select lines, the text is almost unreadable: the font color is nearly the same as the background color.

This is the only file extension that I've encountered up until now where I am not very happy with the looks

**Comment by filipe-maia** [comment]
I can confirm this. I cannot read what I'm writing because the text in the active line is the same colour as the background.

Cursor in the end of line 1:
<img width="219" alt="cursor is in the end of line 1" src="https://user-images.githubusercontent.com/9107130/66040273-f12be780-e50e-11e9-8d85-c27c2fdd09aa.png">

Cursor in the end of line 3:
<img width="540" alt="cursor is in the end of line 3" src="https://user-images.githubusercontent.com/9107130/66040274-f12be780-e50e-11e9-8cf9-8944ac4476ca.png">

Line 3 is selected:
<img width="224" alt="line 3 is selected" src="https://user-images.githubusercontent.com/9107130/66040275-f12be780-e50e-11e9-960d-a4d505497054.png">

Cursor in the end of line 5:
<img width="248" alt="cursor is in the end of line 5" src="https://user-images.githubusercontent.com/9107130/66040276-f12be780-e50e-11e9-911e-82e3306924bd.png">

**Comment by ukarim** [comment]
@vibonacci @filipe-maia Currently npp doesn't support syntax highlighting for markdown files.
But in case of user defined language you can customize xml config that usually placed in `%AppData%\Notepad++\userDefineLangs`.
Just use colors from https://github.com/dracula/dracula-theme#color-palette and replace settings defined in `<Styles>` tag

**Comment by filipe-maia** [comment]
@ukarim Thank you very much.

I'm using a portable version of Notepad++, so I did not go to `%AppData%\Notepad++\userDefineLangs`, instead I went into `{npp portable folder}\userDefineLangs` and there I found a file named _userDefinedLang-markdown.default.modern.xml_.

I opened that file and edited it, I used all the colours in [https://github.com/dracula/dracula-theme#color-palette](https://github.com/dracula/dracula-theme#color-palette) plus a few new ones.

The end result is nothing special, but at least now I can work with MD files without problems:
<img width="288" alt="md" src="https://user-images.githubusercontent.com/9107130/66119198-1e8c9a00-e5d0-11e9-82a7-d4bc817c7c67.png">

Here it is my XML after I edited it:
[userDefinedLang-markdown.default.modern.xml.zip](https://github.com/dracula/notepad-plus-plus/files/3685569/userDefinedLang-markdown.default.modern.xml.zip)

**Comment by GitHub-Mike** [comment]
> Here it is my XML after I edited it:
> [userDefinedLang-markdown.default.modern.xml.zip](https://github.com/dracula/notepad-plus-plus/files/3685569/userDefinedLang-markdown.default.modern.xml.zip)

You have an error in line 37:

`<Styles>bd93f9`

**Comment by zvit** [comment]
I can confirm this when editing an autohotkey file (.ahk)

![notepad++_p2OF5x9w3v](https://user-images.githubusercontent.com/9898388/69255009-c4678880-0bbf-11ea-81f8-94129d3f2f84.png)

**Comment by GitHub-Mike** [comment]
Notepad++ doesn't support syntax highlighting for AutoHotKey files and use global style for it. You need a userDefinedLang XML-file for AHK-files. There you can change the color codes.

**Comment by t1emp0** [comment]
> 
> 
> > Here it is my XML after I edited it:
> > [userDefinedLang-markdown.default.modern.xml.zip](https://github.com/dracula/notepad-plus-plus/files/3685569/userDefinedLang-markdown.default.modern.xml.zip)
> 
> You have an error in line 37:
> 
> `<Styles>bd93f9`
> 
> **And attention the user defined language configuration files are overwritten with the next update!**

Having had this same issue, I ended up in this post.
After reading your last line, that brings up the question if it is worth fixing it or if will get overwritten soon. In other words: is the solution proposed above a permanent solution or more like a temporary fix?

Thank you very much!

**Comment by filipe-maia** [comment]
> You have an error in line 37:
> 
> `<Styles>bd93f9`

Thank you. I've edited my previous post to attach a corrected version of my XML.

**Comment by hnavia** [comment]
> @ukarim Thank you very much.
> 
> I'm using a portable version of Notepad++, so I did not go to `%AppData%\Notepad++\userDefineLangs`, instead I went into `{npp portable folder}\userDefineLangs` and there I found a file named _userDefinedLang-markdown.default.modern.xml_.
> 
> I opened that file and edited it, I used all the colours in https://github.com/dracula/dracula-theme#color-palette plus a few new ones.
> 
> After that, I restarted Notepad++.
> 
> The end result is nothing special, but at least now I can work with MD files without problems:
> <img alt="md" width="288" src="https://user-images.githubusercontent.com/9107130/66119198-1e8c9a00-e5d0-11e9-82a7-d4bc817c7c67.png">
> 
> Here it is my XML after I edited it:
> [userDefinedLang-markdown.default.modern.xml.zip](https://github.com/dracula/notepad-plus-plus/files/3685569/userDefinedLang-markdown.default.modern.xml.zip)
> 
> EDIT: new version of my XML without an error on line 37:
> [userDefinedLang-markdown.default.modern.xml [NEW VERSION, WITHOUT ERROR IN LINE 37].zip](https://github.com/dracula/notepad-plus-plus/files/4180642/userDefinedLang-markdown.default.modern.xml.NEW.VERSION.WITHOUT.ERROR.IN.LINE.37.zip)

Thanks mate, I tried to fix this with no success but your fix did the trick.

**Comment by filipe-maia** [comment]
> Thanks mate, I tried to fix this with no success but your fix did the trick.
No problem. I'm glad it's working for everybody.

**Comment by gildoneto** [comment]
> @ukarim Thank you very much.
> 
> I'm using a portable version of Notepad++, so I did not go to `%AppData%\Notepad++\userDefineLangs`, instead I went into `{npp portable folder}\userDefineLangs` and there I found a file named _userDefinedLang-markdown.default.modern.xml_.
> 
> I opened that file and edited it, I used all the colours in https://github.com/dracula/dracula-theme#color-palette plus a few new ones.
> 
> After that, I restarted Notepad++.
> 
> The end result is nothing special, but at least now I can work with MD files without problems:
> <img alt="md" width="288" src="https://user-images.githubusercontent.com/9107130/66119198-1e8c9a00-e5d0-11e9-82a7-d4bc817c7c67.png">
> 
> Here it is my XML after I edited it:
> [userDefinedLang-markdown.default.modern.xml.zip](https://github.com/dracula/notepad-plus-plus/files/3685569/userDefinedLang-markdown.default.modern.xml.zip)
> 
> EDIT: new version of my XML without an error on line 37:
> [userDefinedLang-markdown.default.modern.xml [NEW VERSION, WITHOUT ERROR IN LINE 37].zip](https://github.com/dracula/notepad-plus-plus/files/4180642/userDefinedLang-markdown.default.modern.xml.NEW.VERSION.WITHOUT.ERROR.IN.LINE.37.zip)

Thank you so much @filipe-maia

**Comment by filipe-maia** [comment]
Good to know the XML I've uploaded is helping :)

**Comment by hnavia** [comment]
It seems it's not working on Notepad++ 8

![image](https://user-images.githubusercontent.com/69368602/127364005-3f911125-5599-43a3-99c4-70a987a70d36.png)

**Comment by hnavia** [comment]
I've used your lines to update the markdown._preinstalled.udl.xml that comes with Notepad++ V8, hope this works for everyone.

All you need to do is replace the file `markdown._preinstalled.udl.xml` located in `%AppData%\Notepad++\userDefineLangs` directory with the one included in the zip file attached.

![image](https://user-images.githubusercontent.com/69368602/127365671-de4b0700-2f5f-4b8e-8d24-e8a23c978160.png)


[markdown._preinstalled.udl.zip](https://github.com/dracula/notepad-plus-plus/files/6895179/markdown._preinstalled.udl.zip)

**Comment by asontu** [comment]
From what I can tell, the `markdown._preinstalled.udl.xml` file gets overwritten with every Notepad++ update, so here is @hnavia's file but altered to be called *Markdown (Dracula)* in the menu and renamed `markdown.dracula.udl.xml` so it won't get overwritten with the next update.

[markdown.dracula.udl.xml.zip](https://github.com/dracula/notepad-plus-plus/files/6911323/markdown.dracula.udl.xml.zip)

**Comment by GitHub-Mike** [comment]
I had skipped some Notepad++ updates in the past because of the problem with overwriting the userDefineLangs directory.

Today I did the update to v8.1.2 and the directory was overwritten again and contained a file markdown._preinstalled.udl.xml afterwards.

Furthermore, there was a problem with the automatic language recognition of the open markdown files. Even after a restarting  notepad++, they were only recognised as none (normal text).  Only after closing and reloading the files were they recognised correctly as markdown again.

Nothing has changed in the udl format. It does not matter what the file name is. Only the ending must be .xml. The only important thing with several existing files is that name and ext have different values:

`<UserLang name="Markdown (Default)" ext="md markdown" udlVersion="2.1">`

Since the problem with overwriting the userDefineLangs directory has not been solved, one must continue to save this and restore it after an update.
